contracts/interfaces/IGeneralOrder.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+struct GeneralOrder {
```

Comment:
nit: It's general enough to call it `Order`.  Other specific type order (if there is in some future 🤣) could then use prefix/suffix to express, like `LimitOrder`, `SpecialOrder` etc.

Reply:
I'll rename this part and update type hash in another PR.